### ISSUE TITLE
Reactivity issue when passing a signal

### ISSUE THREAD

**Issue by OrJDev** [solidjs-community/eslint-plugin-solid#80]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Environment (please complete the following information):**

- OS: [e.g. Mac OS 11, Windows 10]
- Node version (`node --version`):
- `eslint-plugin-solid` version (`npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`):
- `eslint` version (`npm list eslint`/`yarn why eslint`):

**Additional context**

- [ ] I would be willing to contribute a PR to fix this issue

```ts
const addRes = add(
    () => ({
      a: num1(),
      b: 3,
    }),
    () => ({
      placeholderData: (prev) => prev,
      onSettled(data, error) {
        console.log({ data, error })
      },
      onSuccess: (data) => {
        console.log('success', data)
      },
      alwaysCSRRedirect: true,
    })
  )
```

This seems to throw a warning about reactivity, tho it is a perfect valid usecase to pass signals. I previously made an issue about it but seems like it was partially fixed, by partially fixed i mean only when using a deeper function:

add(signal) // error
something.add(signal) // fine

**Comment by OrJDev** [comment]
Previous issue is https://github.com/solidjs-community/eslint-plugin-solid/issues/54

**Comment by joshwilsonvu** [comment]
Assuming `num1` is a signal, it doesn't seem like your example is doing anything unexpected, even if you replace `add()` with `something.add()`. I think the issue is the naming—there's a check like

```js
if (/^(?:use|create)[A-Z]/.test(callee.name)) {
  // be extremely permissive about accepting reactive variables as arguments
}
```

There's not a good way to analyze functions (especially imported functions) for whether they accept reactive variables, act as tracked scopes, etc. So you can pass anything you want to `createAdd(...)` or `something.createAdd(...)`. But with just `add()` here, there's no reason for the plugin to assume that it supports reactivity, so it warns. 

If there's something specific you'd like to change with the plugin, let me know, but I'm going to close this 🙂
